lib/types/api.d.ts

```diff
@@ -0,0 +1,86 @@
+import nodeify from '../utils/nodeify'
+
+interface IOption {
+    hostUrl: string;
+    key_id: string;
+    key_secret?: string;
+    ua: string;
+    headers?: string;
+}
+
+interface IPayload<T> {
+    url: string;
+    data: T;
+}
+
+export type INotify = 'email' | 'sms'
+
+export interface RazorpayHeaders {
+    'X-Razorpay-Account'?: string;
+    'Content-Type'?: string;
+}
+
+/**
+ * Key-value pairs 
+ */
+export interface IMap<T> {
+    [key: string]: T | null;
+}
+
+export type PartialOptional<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
+export type RequiredOptional<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>
+
+export interface RazorpayQuery {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Lets name this RazorpayPaginationOptions